Cargo.toml

```diff
@@ -49,6 +49,7 @@ smallvec = "1"
 tracing = { version = "0.1", default-features = false, features = ["std"] }
 typed-arena = "2.0.1"
 serde = { version = "1", features = ["derive"], optional = true }
+futures = "0.3.9"
```

Comment:
would be quite unfortunate to have this dependency just for this tiny little bit. Can we work around it?